snapshots/snapshots/src/main/kotlin/com/emergetools/snapshots/compose/SnapshotVariantProvider.kt

```diff
@@ -0,0 +1,35 @@
+package com.emergetools.snapshots.compose
+
+import android.content.res.Configuration
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.Density
+import com.emergetools.snapshots.shared.ComposePreviewSnapshotConfig
+import java.util.Locale
+
+@Composable
+fun SnapshotVariantProvider(
+  config: ComposePreviewSnapshotConfig,
+  content: @Composable () -> Unit,
+) {
+  val fontScale = config.fontScale ?: 1.0f
```

Comment:
µ-nit: some of the default handling is duplicated. Eg this gets parsed as:
```
val fontScaleValue = fontScaleArgument?.value?.takeIf {
      (it as? Float) != 1.0f
```
so we only take the value if it's not the default scale magic value, but then we default it right back to the 1.0f value if the config is missing. I see what you're doing by making all of the config properties optional though, so something is needed at this level. Maybe pulling the default values out of this function would make that a little clearer. Very minor so up to you.

Reply:
Great point, updated to only set these if provided!